cartesian_admittance_controller:
  ros__parameters:
    command_interfaces:
      - position
      # - velocity
    state_interfaces:
      - position
      - velocity
    joints:
      - joint_a1
      - joint_a2
      - joint_a3
      - joint_a4
      - joint_a5
      - joint_a6

    filters:
      state_filter_cuttoff_freq: 100.0
      command_filter_cuttoff_freq: 100.0

    kinematics:
      plugin_name: kinematics_interface_kdl/KinematicsInterfaceKDL
      plugin_package: kinematics_interface
      base: base_link # Assumed to be stationary
      tip: interaction_point
      group_name: arm
      alpha: 0.0005

    ft_sensor:
      name: ftsensor
      frame:
        id: ft_sensor  # Wrench measurements are in this frame
        external: false # force torque frame exists within URDF kinematic chain
      filter_coefficient: 0.1 # 1 - exp(-Ts/tau) = 1 - exp(-Ts*2*pi*f_cutoff)

    control:
      frame:
        id: interaction_point # Admittance calcs (displacement etc) are done in this frame. Usually the tool or end-effector
        external: false # control frame exists within URDF kinematic chain

    fixed_world_frame:   # Gravity points down (neg. Z) in this frame (Usually: world or base_link)
      frame:
        id: world # Admittance calcs (displacement etc) are done in this frame. Usually the tool or end-effector
        external: false # control frame exists within URDF kinematic chain

    gravity_compensation:
      frame:
        id: ft_sensor
        external: false

      CoG: # specifies the center of gravity of the end effector (post FT sensor)
        pos:
          - 0.0015 # x
          - 0.0001 # y
          - 0.0371 # z
        force: 0.5 # mass * 9.81

    admittance:
      frame:
        id: interaction_point # Admittance calcs (displacement etc) are done in this frame. Usually the tool or end-effector
      # Admittance rule
      plugin_package: cartesian_admittance_controller
      plugin_name: cartesian_admittance_controller/VanillaCartesianAdmittanceRule  # Nominal controller

      selected_axes:
        - true # x
        - true # y
        - true # z
        - true # rx
        - true # ry
        - true # rz

      # Having ".0" at the end is MUST, otherwise there is a loading error
      # F = M*a + D*v + S*(x - x_d)
      inertia:
        - 5.0
        - 5.0
        - 5.0
        - 0.1
        - 0.1
        - 0.1

      damping_ratio: # damping can be used instead: zeta = D / (2 * sqrt( M * S ))
        - 0.9 # x
        - 0.9 # y
        - 0.9 # z
        - 0.99 # rx
        - 0.99 # ry
        - 0.99 # rz

      stiffness:
        - 200.0
        - 200.0
        - 200.0
        - 2.0
        - 2.0
        - 2.0
